llrt_core/src/vm.rs

```diff
@@ -184,6 +184,28 @@ impl Vm {
 
         self.run(source, strict, global).await;
     }
+
+    pub async fn run_repl(&self) {
+        self.ctx
+            .with(|ctx| -> Result<()> {
+                let global = ctx.globals();
+                let console: Object = global.get("console")?;
+                let js_log: Function = console.get("log")?;
+                loop {
+                    let mut input = String::new();
+                    print!("> ");
+                    std::io::Write::flush(&mut std::io::stdout())?;
+                    std::io::stdin().read_line(&mut input)?;
+                    ctx.eval::<Value, _>(input.as_bytes())
+                        .and_then(|ret| js_log.call::<(Value<'_>,), ()>((ret,)))
+                        .catch(&ctx)
+                        .unwrap_or_else(|err| println!("{err}"));
```

Comment:
I think the right approach here would be to evaluate the code and return value and pass that value to format method and write to std out